x/concentrated-liquidity/tick.go

```diff
@@ -191,6 +191,34 @@ func validateTickRangeIsValid(tickSpacing uint64, lowerTick int64, upperTick int
 	return nil
 }
 
+// roundTick takes a tick and determines if multiple ticks can represent the same price as the provided tick. If so, it
+// rounds that tick up to the largest tick that can represent the same price that the original tick corresponded to. If one of
+// the two ticks are modified, we re-validate the tick range to ensure that the tick range is still valid.
+//
+// i.e. the provided tick is -161795100. With our precision, this tick correlates to a sqrtPrice of 0.000000001414213563
+// the first tick (given our precision) that is able to represent this price is -161000000, so we use this tick instead.
+//
+// This really only applies to very small tick values, as the increment of a single tick continues to get smaller as the tick value gets smaller.
+func roundTick(lowerTick, upperTick int64, priceTickLower, priceTickUpper sdk.Dec, tickSpacing uint64) (int64, int64, error) {
+	newLowerTick, err := math.PriceToTickRoundDown(priceTickLower, tickSpacing)
+	if err != nil {
+		return 0, 0, err
+	}
+	newUpperTick, err := math.PriceToTickRoundDown(priceTickUpper, tickSpacing)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// If the lower or upper tick has changed, we need to re-validate the tick range.
+	if lowerTick != newLowerTick.Int64() || upperTick != newUpperTick.Int64() {
+		err := validateTickRangeIsValid(tickSpacing, newLowerTick.Int64(), newUpperTick.Int64())
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	return newLowerTick.Int64(), newUpperTick.Int64(), nil
```

Comment:
Let's initialize `Int64()` variables and avoid recomputation please

Reply:
done here https://github.com/osmosis-labs/osmosis/pull/5173/commits/b5923f01c5039f7e755790196207846a4f306edc